### PR TITLE
451-RPV-1-tab-buttons

### PR DESCRIPTION
**Pull request by ahjyrkia**
closes https://github.com/HSLdevcom/jore-map-ui/issues/450

### CODE REVIEW

File: src/components/sidebar/routePathView/routePathView.scss [hunk 1]
```diff
@@ -3,22 +3,52 @@
...
-    .staticInfo {
-        margin-top: 5px;
-        font-size: $smallFontSize;
-        margin-right: 10px;
+    .tabButton {
+        width:100%;
+        height:40px;
+        background-color: $lightGrey;
+        border-bottom: none;
+        border-bottom: 1.5px solid $mediumLightGrey;
```
Comment (neodyymi) [inline src/components/sidebar/routePathView/routePathView.scss]: why 2 border-bottoms?

File: src/components/sidebar/routePathView/routePathView.scss [hunk 2]
```diff
@@ -3,22 +3,52 @@
...
+        color: black;
+        font-weight: 500;
+        border-bottom: none;
+    }
+
+    .topic {
+        padding: 10px;
         font-weight: bold;
+        color: #898989;
+        font-size: 1rem;
```
Comment (neodyymi) [inline src/components/sidebar/routePathView/routePathView.scss]: Would be cleaner if both border bottom and side would be on same item.
Now the corner is empty when rightmost tab is selected.

File: src/components/sidebar/routePathView/NewRoutePathView.tsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ import { ToolbarStore } from '~/stores/toolbarStore';
 import RoutePathFactory from '~/factories/routePathFactory';
 import ViewHeader from '../ViewHeader';
 import RoutePathViewForm from './RoutePathViewForm';
-import * as s from './routePathView.scss';
+import * as s from './routePathViewTab.scss';
```
[the remark this answers is not in the document]
Comment (ahjyrkia) [reply src/components/sidebar/routePathView/NewRoutePathView.tsx]: fixed

File: src/components/sidebar/routePathView/RoutePathView.tsx [hunk 4]
```diff
@@ -62,143 +53,56 @@ class RoutePathView extends React.Component<IRoutePathViewProps, IRoutePathViewS
...
-    }
-
-    public onChange = (property: string, value: any, validationResult?: IValidationResult) => {
-        this.setState({
-            routePath: { ...this.state.routePath!, [property]: value },
-            hasModifications: true,
+    public generateTabs = () => {
+        return tabs.map((tab: string, index) => {
+            return(
+                <div
```
[the remark this answers is not in the document]
Comment (ahjyrkia) [reply src/components/sidebar/routePathView/RoutePathView.tsx]: Made it into its own component

File: src/components/sidebar/routePathView/routePathView.scss (hunk 2 above)
Comment (ahjyrkia) [reply src/components/sidebar/routePathView/routePathView.scss]: fixed

File: src/components/sidebar/routePathView/RoutePathView.tsx [hunk 5]
```diff
@@ -62,143 +45,43 @@ class RoutePathView extends React.Component<IRoutePathViewProps, IRoutePathViewS
...
-                    <Button
-                        onClick={this.toggleEditing}
-                        type={ButtonType.SQUARE}
-                        text={this.state.isEditingDisabled ? 'Muokkaa' : 'Peruuta'}
+            <div className={s.routePathView}>
+                <div className={s.flexInnerRow}>
+                    <RoutePathViewTabButtons
+                        tabs={['Reitinsuunta', 'Solmut ja linjat']}
+                        selectedTab={this.state.selectedTab}
+                        onClick={this.onTabClick}
```
Comment (Divinare) [inline src/components/sidebar/routePathView/RoutePathView.tsx]: selectTab

Comment (neodyymi) [reply src/components/sidebar/routePathView/RoutePathView.tsx]: Isn't this exactly `selectedTab` and not `selectTab`?

Comment (ahjyrkia) [reply src/components/sidebar/routePathView/RoutePathView.tsx]: fixed

File: src/components/sidebar/routePathView/RoutePathViewTab.tsx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+    private redirectToNewRoutePathView = () => {
+        const routePath = this.state.routePath;
+        if (!routePath) return;
+
+        const newRoutePathLink = routeBuilder
+        .to(subSites.newRoutePath, { routeId: routePath.routeId, lineId: routePath.lineId })
+        .toLink();
+
+        this.props.routePathStore!.setRoutePath(this.state.routePath);
```
[the remark this answers is not in the document]
Comment (neodyymi) [reply src/components/sidebar/routePathView/RoutePathViewTab.tsx]: Could we have it as an issue then?

File: src/components/sidebar/routePathView/RoutePathView.tsx [hunk 7]
```diff
@@ -62,143 +53,56 @@ class RoutePathView extends React.Component<IRoutePathViewProps, IRoutePathViewS
         });
     }
 
-    public toggleEditing = () => {
-        const isEditingDisabled = !this.state.isEditingDisabled;
-        this.setState({ isEditingDisabled });
-    }
-
-    public save = async () => {
```
[the remark this answers is not in the document]
Comment (neodyymi) [reply src/components/sidebar/routePathView/RoutePathView.tsx]: Is there an issue to deal with it?

File: src/components/sidebar/routePathView/RoutePathView.tsx (hunk 5 above)
Comment (neodyymi) [reply src/components/sidebar/routePathView/RoutePathView.tsx]: Ah... You meant the onClick

File: src/components/sidebar/routePathView/RoutePathView.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (ahjyrkia) [reply src/components/sidebar/routePathView/RoutePathView.tsx]: Not yet no but Joe is on it

File: src/components/sidebar/routePathView/RoutePathView.tsx [hunk 8]
```diff
@@ -1,49 +1,32 @@
...
+import RoutePathViewTabButtons from './RoutePathViewTabButtons';
 import * as s from './routePathView.scss';
 
 interface IRoutePathViewState {
-    isEditingDisabled: boolean;
-    hasModifications: boolean;
     routePath: IRoutePath | null;
     isLoading: boolean;
-    invalidFieldsMap: object;
+    selectedTab: string;
```
Comment (kronken) [inline src/components/sidebar/routePathView/RoutePathView.tsx]: make enum

File: src/components/sidebar/routePathView/RoutePathView.tsx [hunk 9]
```diff
@@ -62,143 +45,43 @@ class RoutePathView extends React.Component<IRoutePathViewProps, IRoutePathViewS
...
-                    closePromptMessage={this.state.hasModifications ? message : undefined}
-                >
-                    <Button
-                        onClick={this.toggleEditing}
-                        type={ButtonType.SQUARE}
-                        text={this.state.isEditingDisabled ? 'Muokkaa' : 'Peruuta'}
+            <div className={s.routePathView}>
+                <div className={s.flexInnerRow}>
+                    <RoutePathViewTabButtons
+                        tabs={['Reitinsuunta', 'Solmut ja linjat']}
```
Comment (kronken) [inline src/components/sidebar/routePathView/RoutePathView.tsx]: What about 'Reitinsuunnan tiedot'?

File: src/components/sidebar/routePathView/RoutePathView.tsx [hunk 10]
```diff
@@ -62,143 +45,43 @@ class RoutePathView extends React.Component<IRoutePathViewProps, IRoutePathViewS
...
-                        <Moment
-                            date={this.state.routePath.lastModified}
-                            format='DD.MM.YYYY HH:mm'
-                        />
-                    </div>
-                    <div className={s.flexInnerColumn}>
-                        <div>Päivittäjä</div>
-                        <div>{this.state.routePath.modifiedBy}</div>
+
+                {(this.state.selectedTab === 'Reitinsuunta') ?
```
Comment (neodyymi) [inline src/components/sidebar/routePathView/RoutePathView.tsx]: This feels sort of magic numberish. Should it rather be a component specified by key?

File: src/components/sidebar/routePathView/RoutePathView.tsx (hunk 9 above)
Comment (Divinare) [reply src/components/sidebar/routePathView/RoutePathView.tsx]: For me "Reitinsuunta" is fine, its in sync with "Solmut ja linjat"

File: src/components/sidebar/routePathView/RoutePathView.tsx (hunk 8 above)
Comment (ahjyrkia) [reply src/components/sidebar/routePathView/RoutePathView.tsx]: done

File: src/components/sidebar/routePathView/RoutePathView.tsx (hunk 10 above)
Comment (ahjyrkia) [reply src/components/sidebar/routePathView/RoutePathView.tsx]: I don't know if there's a better solution for this. With enums it's maybe a bit less magic numberish now

File: src/components/sidebar/routePathView/RoutePathView.tsx [hunk 11]
```diff
@@ -64,13 +65,13 @@ class RoutePathView extends React.Component<IRoutePathViewProps, IRoutePathViewS
             <div className={s.routePathView}>
                 <div className={s.flexInnerRow}>
                     <RoutePathViewTabButtons
-                        tabs={['Reitinsuunta', 'Solmut ja linjat']}
+                        tabs={[RoutePathViewTabType.ROUTE, RoutePathViewTabType.NODES_AND_LINES]}
```
Comment (neodyymi) [inline src/components/sidebar/routePathView/RoutePathView.tsx]:
```
const tabs = [{tab: RoutePathViewTabType.Route, content: <RoutePathViewTab routePath={this.state.routePath}}, {tab:..., content:...}]
```
Could be better to have this list dynamic for both tabs generated and view shown.

Comment (neodyymi) [reply src/components/sidebar/routePathView/RoutePathView.tsx]: Or note this as a todo if there's a hurry to get this before demo

File: src/components/sidebar/routePathView/RoutePathView.tsx [hunk 12]
```diff
@@ -62,143 +46,43 @@ class RoutePathView extends React.Component<IRoutePathViewProps, IRoutePathViewS
...
-                    <div className={s.flexInnerColumn}>
-                        <div>Päivittäjä</div>
-                        <div>{this.state.routePath.modifiedBy}</div>
+
+                {(this.state.selectedTab === RoutePathViewTabType.ROUTE) ?
+                    <RoutePathViewTab
+                        routePath={this.state.routePath}
+                    /> :
+                    <div>
+                        Solmut ja linjat
```
Comment (ahjyrkia) [inline src/components/sidebar/routePathView/RoutePathView.tsx]: Solmut ja linkit*

**Pull request merged**